### DISCUSSION TITLE
two different sets of pages, how to organize side bar

### DISCUSSION THREAD

**Discussion by hasakilol** [umijs/umi#11982] [General]
umi version: **umi 4**

My website have 2 different sets of pages: Page Set 1 and Page Set 2.

**Page Set 1** and **Page Set 2**  both have a sidebar: Side bar 1 and Side bar 2.

If I put all pages into `routes.config.ts`, then items of Side bar 1 and Side bar 2 are all merged together, am I right? If so, what is the recommended way to separate these items into 2 side bar?

**Comment by hasakilol** [comment]
Solution 1: Implement 2 apps: one for Page Set 1, the other for Page Set 2
Solution 2: in initial state, decide whether current page belongs to Page Set 1 or Page Set 2. At the same time, in `routes.config.ts`, for each route add `access: 'belongToPageSet1'` or `access: 'belongToPageSet2'`

**Comment by fz6m** [comment]
1. 用 [`patchClientRoutes`](https://umijs.org/docs/api/runtime-config#patchclientroutes-routes-) 在运行时动态添加路由，可参考 https://github.com/umijs/umi/discussions/9717#discussioncomment-4078053 。

2. 你上文提到的用 access 去过滤也可以。
